g-sonde/src/pages/User/Profile.tsx

```diff
@@ -0,0 +1,326 @@
+                    },
+                    {
+                        withCredentials: true,
+                    }
+                )
+                .then(() => {
+                    setShowSuccessToast(true);
+                    setToastMessage("Profile picture updated successfully");
+                })
+                .catch(() => {
+                    setShowErrorToast(true);
+                    setToastMessage("Error updating profile picture");
+                });
+        };
+        reader.readAsDataURL(file);
+    };
+
+    useIonViewDidEnter(getProfilePicture);
+
+    return (
+        <IonPage>
+            <IonHeader>
+                <IonToolbar>
+                    <IonTitle>My Profile</IonTitle>
+                </IonToolbar>
+            </IonHeader>
+            <IonContent className="ion-padding">
+                <div className="md:w-[35em] md:m-auto pt-20">
+                    <h1 className="text-4xl text-center mb-6">
+                        Modify your profile
+                    </h1>
+                    <img
+                        id="profile-picture"
+                        className="rounded-full w-32 h-32 mx-auto my-10"
+                        src="/no-profile-picture.png"
+                        alt="profile picture"
+                    ></img>
+                    <div className="flex justify-center ">
+                        <label
+                            htmlFor="input-file"
+                            className="border border-blue-400 p-3 rounded-md cursor-pointer hover:bg-blue-400 hover:text-white duration-150 ease-in-out"
+                        >
+                            Change your profile picture
+                        </label>
+                        <input
+                            id="input-file"
+                            accept="image/jpeg, image/png, image/jpg"
+                            type="file"
+                            onChange={handleProfilePicture}
+                            className="hidden"
+                        />
+                    </div>
+                    <IonCard className="">
+                        <IonCardContent>
+                            <form id="main-form">
+                                <div className="flex flex-col md:flex-row">
+                                    <IonItem className="w-[50%]">
+                                        <IonLabel position="floating">
+                                            Firstname
+                                        </IonLabel>
+                                        <IonInput
+                                            type="email"
+                                            onIonChange={(e) =>
+                                                setFirstname(e.detail.value!)
+                                            }
+                                            required
+                                        ></IonInput>
```

Comment:
[Ionic Warning]: ion-input now requires providing a label with either the "label" property or the "aria-label" attribute. To migrate, remove any usage of "ion-label" and pass the label text to either the "label" property or the "aria-label" attribute.

Example: <ion-input label="Email"></ion-input>
Example with aria-label: <ion-input aria-label="Email"></ion-input>

For inputs that do not render the label immediately next to the input, developers may continue to use "ion-label" but must manually associate the label with the input by using "aria-labelledby".

Developers can use the "legacy" property to continue using the legacy form markup. This property will be removed in an upcoming major release of Ionic where this form control will use the modern form markup.